community-stabilization-fund-api/pages/admin/users/[id].tsx

```diff
@@ -16,15 +19,23 @@ const AdminPage = () => {
     id: 0,
     name: '',
   });
+  const router = useRouter();
   const inputRef = useRef('');
   useEffect(() => {
-    const getUser = async () => {
-      const user = await UserService.getById(id as string);
-      setUser(user?.data[0]);
-    };
-
-    getUser();
-  }, [id]);
+    if (!!id) {
+      router.push('/admin/login');
+    } else {
+      const getUser = async () => {
+        const user = await UserService.getById(id as string);
+        if (typeof user === 'undefined') {
+          router.push('/admin/login');
+        } else {
```

Comment:
Didn't notice you were explicitly redirecting twice. Let's not do this. Instead, to add a guard case, just prevent the function getUser from returning values if the localStorage isn't set to a string and then only call setUser IF `user?.data`exists. The RouteGuard should be handling redirects. We just got to make sure setUser isn't accidentally calling when there's no `id` or `user`. 

`if(!id || typeof id !== 'string') return;`

`if (user?.data) setUser(user?.data[0]);`